Wikipedia/View Controllers/WebView/NativeSectionTitleOverlays/WMFSectionTitlesViewController.m

```diff
@@ -0,0 +1,231 @@
+//  Created by Monte Hurd on 7/31/15.
+//  Copyright (c) 2015 Wikimedia Foundation. Provided under MIT-style license; please copy and modify!
+
+#import "WMFSectionTitlesViewController.h"
+#import "WMFTitleOverlayLabel.h"
+#import "WMFTitleOverlayModel.h"
+#import "NSString+Extras.h"
+#import "UIView+WMFSearchSubviews.h"
+#import <Masonry/Masonry.h>
+
+@interface WMFSectionTitlesViewController ()
+
+@property (nonatomic, strong) NSMutableArray* models;
+@property (nonatomic) NSUInteger indexOfNativeTitleLabelNearestTop;
+@property (nonatomic, strong) NSLayoutConstraint* topStaticNativeTitleLabelTopConstraint;
+@property (nonatomic, strong) WMFTitleOverlayLabel* topStaticNativeTitleLabel;
+
+@end
+
+@implementation WMFSectionTitlesViewController
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        self.models                            = @[].mutableCopy;
+        self.indexOfNativeTitleLabelNearestTop = -1;
+    }
+    return self;
+}
+
+- (void)hideTopOverlay {
+    self.topStaticNativeTitleLabel.alpha = 0;
+}
+
+- (void)addOverlays {
```

Comment:
I think something like `reload` or `reset` would make more sense here instead of `add` (which implies that the operation isn't idempotent and doesn't **remove** anything)